File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
        int ans = 0;
        int n = heights.length;
        int left = 0;
        int right = n-1;
        while(left<right){
            ans = Math.max(ans,(right-left)*Math.min(heights[left],heights[right]));
            if(heights[left]<heights[right])left++;
            else right--;
        }
        return ans;
    }
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int left = 0;
        int right = 0;
        int ans = nums.size();
        int currentSum =0;
        while(right<nums.size()){
            if(currentSum<target){
                currentSum+=nums[right];
                right++;
            }
            else{
                ans = min(ans,right-left);
                currentSum -= nums[left];
                left++;
            }
        }
        while(left<=right && currentSum>=target){
            ans = min(ans,right-left);
            currentSum -= nums[left];
            left++;
        }
        if(currentSum<target && left==0)return 0;
        return ans;
    }
};